Qast/Protocols/SoundZoneAPI.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  SoundZoneAPI.swift
+//  Qast
+//
+//  Created by Austin Welch on 7/7/19.
+//  Copyright © 2019 Qast. All rights reserved.
+//
+
+import CoreLocation.CLLocation
+
+typealias SoundZoneCompletionBlock = ([SoundZone], Error?) -> Void
+
+protocol SoundZoneAPI {
+    func soundZones(nearby location: CLLocationCoordinate2D, distance: Double, _ completion: @escaping SoundZoneCompletionBlock)
+    
+    func getSoundZone(id: String, completion: @escaping (Result<SoundZone>) -> Void)
+    func createSoundZone(_ soundZone: SoundZone, completion: @escaping (Result<SoundZone>) -> Void)
+}
+
+extension SoundZoneAPI {
```

Comment:
As a protocol SoundZoneAPI itself shouldn't have any method implementations on it.

Just an interface describing all possible CRUD operations an implementing class (like FirebaseManager) can perform with the SoundZone domain object

Reply:
Do you not like the extension here? The reason for using this extension is to offer an optional implementation of these functions. This way, the confirming class isn't forced to implement these functions

Reply:
Ah I see, I didn't know how to make protocol implementations optional.

It's sometimes misleading to have optional repo methods like this, because if you were to switch out a backend for another implementing class that lacks certain methods, you're gonna be getting back a bunch of nils etc. depending on how that person choose to interpret the repo protocol.

So the non-optional repo methods are meant to provide a 100% consistent API access layer across your app.